### ISSUE TITLE
Delay between actions

### ISSUE THREAD

**Issue by lionB777** [iharosi/homebridge-plex-webhooks#11]
hi,
Is there any chance to have 
-a delay fonction ?  to avoid lights to turn on between  clips and movies ( and between clips ). 
-a delay fonctions : to allow lights to turn on after a given delay.
-an option to inhibit actions if events appends within a short delay ?

thanks for considering.

**Comment by iharosi** [comment]
Hi @lionB777,

I think a simple delay function could be doable as a config option for a sensor.
However, I'm not sure if I understand you well, when you say „inhibit actions if events appends within a short delay”. You mean if one of a sensor has a 10 sec delay and triggered meanwhile a second sensor triggered without delay the latter should terminate the former action which should have ran after? Could you please give me a living example? Or what is the exact problem what you were facing with?

I have a clip sensor for [Plex Pre-Rolls](https://prerolls.video) to dim the lights then I'm using a movie sensor to turn off the lights completely. In my case, the clip sensor does not have a „Stops Detecting Occupancy” automation in Home app because the second sensor will turn off the lights completely, so there won't be any overlap.

**Comment by lionB777** [comment]
Hi @iharosi 
Thanks for your quick return. 
What I meant by inhibit action: 
-in my Plex configuration, when I play a movie, I've got 2 clips in front of a movie, 
-in my Homekit automations I made an event when a clip stops playing, when a movie stop playing too.
my plan was to dim down lights when a clips plays and to Switch off light when a movie play, on any pause event to have a light switching on in the living room corner. On any stop event Light goes back to full
But this what happening actually : 
button play pressed  
- light dim down while 1st clip playing (if I hit pause light switch on in the corner as expected )
- while the 2nd clip ( or a movie ) is about to kick in  the light goes back to full for about 0,5 sec.
So I'd like to avoid this flash , while I will still have the feature of light going back to full on a stop event during clips playing. So I was thinking about this inhibit stuff as it could be some kind of buffer between clip changes to avoid quick changes of sensor status 
Thinking deeper about my request I realize that this buffer mays not work between clip and movies as I change sensor. But for this I was thinking about something who would allow it to work: this inhibit feature would allow the clip sensor to stay on avoiding this flash as Homekit would trigger a new event when a sensor will change state.
I hope I was more clear. thanks
